services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py

```diff
@@ -304,16 +319,26 @@ def get_dynamic_sidecar_spec(
             dynamic_services_scheduler_settings.DYNAMIC_SIDECAR_PROMETHEUS_SERVICE_LABELS,
             scheduler_data.callbacks_mapping,
         )
-        | StandardSimcoreDockerLabels(
-            user_id=scheduler_data.user_id,
-            project_id=scheduler_data.project_id,
-            node_id=scheduler_data.node_uuid,
-            product_name=scheduler_data.product_name,
-            simcore_user_agent=scheduler_data.request_simcore_user_agent,
-            swarm_stack_name=dynamic_services_scheduler_settings.SWARM_STACK_NAME,
-            memory_limit=ByteSize(0),  # this should get overwritten
-            cpu_limit=0,  # this should get overwritten
-        ).to_simcore_runtime_docker_labels(),
+        | standard_simcore_docker_labels
+    )
+
+    # add autoscaling constraints if pricing plan is required
+    if hardware_info and len(hardware_info.aws_ec2_instances) > 0:
+        if len(hardware_info.aws_ec2_instances) == 1:
```

Comment:
THOUGHT: Can you validate earlier `aws_ec2_instances`? i.e. it seems to me that this is a guarantee that should be provided by `hardware_info`. For instance, when you get `hardward_info` from RUT you do this check and if it fails you log a warning (as in here) and set `hardward_info = INVALID` or `None`